packages/@react-aria/numberfield/src/useNumberField.ts

```diff
@@ -11,21 +11,31 @@
  */
 
 import {AriaButtonProps} from '@react-types/button';
-import {AriaNumberFieldProps} from '@react-types/numberfield';
-import {filterDOMProps, mergeProps, useId} from '@react-aria/utils';
-import {HTMLAttributes, LabelHTMLAttributes, RefObject, useEffect, useState} from 'react';
+import {clearAnnouncer} from '@react-aria/live-announcer';
+import {
+  HTMLAttributes,
+  LabelHTMLAttributes,
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react';
 // @ts-ignore
 import intlMessages from '../intl/*.json';
+import {mergeProps, useId} from '@react-aria/utils';
+import {NumberFieldProps} from '@react-types/numberfield';
 import {NumberFieldState} from '@react-stately/numberfield';
-import {SpinButtonProps, useSpinButton} from '@react-aria/spinbutton';
-import {useFocus} from '@react-aria/interactions';
-import {useMessageFormatter} from '@react-aria/i18n';
+import {useFocus, useFocusWithin} from '@react-aria/interactions';
+import {useLocale, useMessageFormatter, useNumberFormatter} from '@react-aria/i18n';
+import {useSpinButton} from '@react-aria/spinbutton';
 import {useTextField} from '@react-aria/textfield';
 
-interface NumberFieldProps extends AriaNumberFieldProps, SpinButtonProps {
-  decrementAriaLabel?: string,
-  incrementAriaLabel?: string
-}
+/**
+ * From https://github.com/filamentgroup/formcore/blob/master/js/numeric-input.js#L20.
```

Comment:
Ah no, because we aren't using it, the note is there for anyone who comes along and tries to change the input type to number. I can move it to a more relevant spot